File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        
        int sell = 1;
        int buy = 0;
        int ans = 0;

        while(sell < n) {
            if(prices[sell] > prices[buy]) {
                ans = max(ans, prices[sell] - prices[buy]);
            }
            else {
                buy = sell;
            }

            sell++;
        }

        return ans;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();

        int p1 = 0;
        int p2 = 0;

        HashSet<Character> set = new HashSet<>();
        int ans = 0;

        while(p1 < n && p2 < n) {

            if(set.contains(s.charAt(p2))) {
                set.remove(s.charAt(p1));
                p1++;
                
            } else {
                set.add(s.charAt(p2));
                ans = Math.max(ans, (p2 - p1) + 1);
                 p2++;
            }
           
        }

        return ans;
    }
}